clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/AbstractGoogleServerGroupCachingAgent.java

```diff
@@ -887,6 +894,77 @@ private static void populateAutoscaler(
     }
   }
 
+  private static GoogleAutoscalingPolicy convertAutoscalingPolicy(AutoscalingPolicy input) {
+    CpuUtilization cpu = convertCpuUtilization(input.getCpuUtilization());
+    LoadBalancingUtilization loadBalancing =
+        convertLoadBalancingUtilization(input.getLoadBalancingUtilization());
+    List<CustomMetricUtilization> customMetrics =
+        convertCustomMetricUtilizations(input.getCustomMetricUtilizations());
+    GoogleAutoscalingPolicy output = new GoogleAutoscalingPolicy();
+    output.setCoolDownPeriodSec(input.getCoolDownPeriodSec());
+    output.setCpuUtilization(cpu);
+    output.setCustomMetricUtilizations(customMetrics);
+    output.setLoadBalancingUtilization(loadBalancing);
+    output.setMaxNumReplicas(input.getMaxNumReplicas());
+    output.setMinNumReplicas(input.getMinNumReplicas());
+    output.setMode(valueOf(AutoscalingMode.class, input.getMode()));
+    return output;
+  }
+
+  @Nullable
+  private static CpuUtilization convertCpuUtilization(
+      @Nullable AutoscalingPolicyCpuUtilization input) {
+    if (input == null) {
+      return null;
+    }
+    CpuUtilization output = new CpuUtilization();
+    output.setUtilizationTarget(input.getUtilizationTarget());
+    return output;
+  }
+
+  @Nullable
+  private static LoadBalancingUtilization convertLoadBalancingUtilization(
+      @Nullable AutoscalingPolicyLoadBalancingUtilization input) {
+    if (input == null) {
+      return null;
+    }
+    LoadBalancingUtilization output = new LoadBalancingUtilization();
+    output.setUtilizationTarget(input.getUtilizationTarget());
+    return output;
+  }
+
+  @Nullable
+  private static List<CustomMetricUtilization> convertCustomMetricUtilizations(
```

Comment:
Maybe return an `ImmutableList` here to indicate to the caller it can't be modified?  (Or perhaps not as important for private methods.)

Reply:
:+1: